src/nupic/types/Sdr.cpp

```diff
@@ -282,15 +282,14 @@ namespace sdr {
     }
 
     void SparseDistributedRepresentation::randomize(Real sparsity, Random &rng) {
-        NTA_ASSERT( sparsity >= 0.0f and sparsity <= 1.0f );
+        NTA_ASSERT( sparsity >= 0.0f and sparsity <= 1.0f ); //TODO make this only >0 , <1 as the extremes don't make sense
```

Comment:
I think that the extremes do make sense. Both 0 & 100% are valid sparsities.

Reply:
does it make sense to randomize a SDR with 0/100 sparsity?

Reply:
Yes, it makes sense.  The set of SDRs with sparsity 0 contains exactly 1 SDR, and `sdr.randomize(0)` will return a random element from that set.

Reply:
randomize(0.0) will return an empty set {}, I'm not sure we can consider it a SDR. Anyway, removed the comment, and the bounds are already inclusive <=, >=